### PR TITLE
Allow `{lang}` as a placeholder in the skeleton directory

### PR DESCRIPTION
**Pull request by nickvergessen**
Signed-off-by: Joas Schilling <coding@schilljs.com>

### CODE REVIEW

File: lib/private/legacy/util.php [hunk 1]
```diff
@@ -379,7 +379,20 @@ public static function getUserQuota($userId) {
...
+
+		if (!file_exists($skeletonDirectory)) {
+			$dialectStart = strpos($userLang, '_');
+			if ($dialectStart !== false) {
+				$skeletonDirectory = str_replace('{lang}', substr($userLang, 0, $dialectStart), $plainSkeletonDirectory);
+			}
+			if ($dialectStart === false || !file_exists($skeletonDirectory)) {
+				$skeletonDirectory = str_replace('{lang}', 'en', $plainSkeletonDirectory);
+			}
+		}
```
Comment (MorrisJobke) [inline lib/private/legacy/util.php]: Please document this in config.sample.php

Comment (nickvergessen) [reply lib/private/legacy/util.php]: Done

**Review by rullzer — APPROVED** [review]

File: config/config.sample.php [hunk 2]
```diff
@@ -242,6 +242,9 @@
  * The directory where the skeleton files are located. These files will be
  * copied to the data directory of new users. Leave empty to not copy any
  * skeleton files.
+ * ``{lang}`` can be used as a placeholder for the language of the user.
+ * If the directory does not exist, it falls back to non dialect (from ``de_DE``
+ * to ``de``). If that does not exist either, it falls back to ``en``
```
Comment (blizzz) [inline config/config.sample.php]: shouldn' t we fall back to 'c' instead? which can follow organization's default language.

Comment (MorrisJobke) [reply config/config.sample.php]: What is `c`?

Comment (blizzz) [reply config/config.sample.php]: Originally, this comes from  UNIX and defines a simple, default locale. Now, transferred to our use case: if we make 'en' as fallback default,  the using org would need to create an english skeleton folder, or misuse it and place content in its native language if there is no desire or use case to create English content.

Example: you have a Japanese org with branch offices in say Korea and China. They would create the folders 'ja', 'zh' and 'ko'. To satisfy the fallback they would either copy Japanes files into  'en' (or symlink to 'ja'). Which makes it factually wrong. Symlinking 'c' to 'ja' makes more sense and is perhaps less confusing. 

Extended example:  perhaps this org has even a branch office in an english speaking country, but the org's lingua franca is Japanese and 'ja' should be the fallback over 'en'. Now 'c' comes in really handy to steer the default fallback.

In the end it is a detail, but changing 'en' to 'c' isn't a big deal, whether this scenario ever comes to life or not.

Comment (blizzz) [reply config/config.sample.php]: Or just call it 'default' if it's not intuitive enough.

Comment (MorrisJobke) [reply config/config.sample.php]: Makes sense 👍

Comment (MorrisJobke) [reply config/config.sample.php]: Fixed.

**Comment by MorrisJobke** [comment]
@blizzz I added another check if the directory exists and reset it to `''` otherwise to not cause problems during the copy later.

I also changed the name to "default". Should be good now.

**Review by blizzz — APPROVED** [review]
tested successfully

**Pull request merged**